### ISSUE TITLE
OOPS src/unify.c 1267

### ISSUE THREAD

**Issue by flexoron** [trealla-prolog/trealla#203]
v2.20.8
```
?- B = a([B]), A = [a([B])], A = [a(A)].
   B = a([B]), A = [a([a([B])])].

?- B = a([B]), A = [a(A)], A = [a([B])].  % A = ... swapped
*** OOPS src/unify.c 1267
   error(representation_error(term),(=)/2).
```
Swapping around continued
```
?- A = [a(A)], A = [a([B])], B = a([B]).
   A = [a([a([a(...)])])], B = a([a([a([a(...)])])]).

?- A = [a(A)], B = a([B]), A = [a([B])].
*** OOPS src/unify.c 1267
   error(representation_error(term),(=)/2).
```

**Comment by flexoron** [comment]
Don't know if the issue is related to #202
```
?- A=B*A, A=B*B*B.
*** OOPS src/unify.c 1267
   error(representation_error(term),(=)/2).

?- A=B*B*B, A=B*A.
   A = B*B*(B*B)*(B*B), B = A. % expected?
```

**Comment by flexoron** [comment]
v2.20.10
Nothing has changed. (Well, 1267 now 1262)
```
?- B = a([B]), A = [a(A)], A = [a([B])]. 
*** OOPS src/unify.c 1262
   error(representation_error(term),(=)/2).

?- A = [a(A)], B = a([B]), A = [a([B])].
*** OOPS src/unify.c 1262
   error(representation_error(term),(=)/2).
```

**Comment by infradig** [comment]
Sorry I had to revert the fix for that.

**Comment by infradig** [comment]
OOPS fixed, output same.

**Comment by flexoron** [comment]
Fixed with v2.20.13. Many thanks!
```
?- B = a([B]), A = [a([B])], A = [a(A)], A=[B].
   B = a([B]), A = [a([a([B])])].

?- B = a([B]), A = [a(A)], A = [a([B])], A=[B].
   B = a([B]), A = [a([a([a(...)])])].

?- A = [a(A)], A = [a([B])], B = a([B]), A=[B].
   A = [a([a([a(...)])])], B = a([a([a([a(...)])])]).

?- A = [a(A)], B = a([B]), A = [a([B])], A=[B].
   A = [a([a([a(...)])])], B = a([B]).
```
